### ISSUE TITLE
Signs do not regenerate properly with soft-restore

### ISSUE THREAD

**Issue by GabeCurran** [garbagemule/MobArena#772]
# Bug report

**Short description**

With soft-restore on, destroyed signs lose their contents.

**Reproduction steps**

1. Enable soft-restore
2. Place a sign with text on it
3. Destroy sign while playing arena
4. Check to see if sign has text

**Details**

* **MobArena version**: 0.107
* **Server version**: Paper-182, MC: 1.20.1
* **Stacktrace** (if applicable): N/A

**Additional info**

Tried without any plugins besides MobArena and MultiVerse-Core
Video in MA Discord:
https://discord.com/channels/363735561179103237/436983158416867341/1151988475504103494

**Comment by garbagemule** [comment]
I've just confirmed this to be the case (paper-1.20.1-194, MobArena 0.107).

MobArena does correctly register the sign being broken and stores its state during the block break event, and it also correctly repairs the sign again by setting the contents back into the sign when the session ends. This code was written on August 14 in _2011_ (!!) and the logic hasn't changed since then, so this is definitely a case of some minute implementation detail in the server (Spigot/Paper) suddenly changing the behavior of the API that MobArena uses here.

The solution appears to just be a matter of calling `update()` on the sign after setting the contents. Hit me up on Discord if you'd like try out a build with that one extra line in the code 😛

I should mention that I've managed to reproduce the NullPointerException referenced here: https://github.com/garbagemule/MobArena/issues/763#issuecomment-1567335918. I think I have a solution for it, and I'll be sure to include that in a build so you don't get broken arenas. The scary part is that I can actually reproduce that NullPointerException by simply _logging_ the sign state, i.e. by calling its `toString()` method. This is for sure a bug in the API and something that should have been guarded against one way or another. I think I know how to get around it, but I'll have to tinker a bit.

For (my own) reference, here's a stacktrace that I got by attempting to `System.out.println()` the block state data (the "Attachable"):

```
[22:42:33] [Server thread/ERROR]: Could not pass event EntityExplodeEvent to MobArena v0.107.1-SNAPSHOT
java.lang.NullPointerException: Cannot invoke "org.bukkit.block.BlockFace.getOppositeFace()" because the return value of "org.bukkit.material.Sign.getAttachedFace()" is null
        at org.bukkit.material.Sign.getFacing(Sign.java:134) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.material.Sign.toString(Sign.java:233) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at java.lang.String.valueOf(String.java:4220) ~[?:?]
        at java.io.PrintStream.println(PrintStream.java:1047) ~[?:?]
        at com.garbagemule.MobArena.repairable.RepairableAttachable.<init>(RepairableAttachable.java:22) ~[MobArena.jar:?]
        at com.garbagemule.MobArena.repairable.RepairableSign.<init>(RepairableSign.java:12) ~[MobArena.jar:?]
        at com.garbagemule.MobArena.ArenaListener.onEntityExplode(ArenaListener.java:477) ~[MobArena.jar:?]
        at com.garbagemule.MobArena.listeners.MAGlobalListener.entityExplode(MAGlobalListener.java:196) ~[MobArena.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor27.execute(Unknown Source) ~[?:?]
```

The error is slightly different, but it points to [line 134](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/material/Sign.java#134) in the `getFacing()` method, which starts a method chain where the first call to `getAttachedFace()` [can return `null`](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/material/Sign.java#66) if the underlying data value is e.g. 0, which I assume is the case here. _How_ it ends up not being a sound value is beyond me. It appears to happen only to wall signs, which should have exactly 4 different directions (data values), which means it shouldn't ever become `null` under normal circumstances. The material data API _is_ deprecated for removal, however, so it would make sense to try to convert to the new block data API.
